### ISSUE TITLE
amp carousel not working correctly when rtl is set

### ISSUE THREAD

**Issue by omar-dulaimi** [ampproject/amphtml#28692]
## What's the issue?
After multiple days of triage, it seems that setting rtl doesn't work properly.
While it does change the direction, it doesn't actually load all images, just the first ones.
In our case it was a carousel of 4 images, only the first two were loaded!

## How do we reproduce the issue?
To try it out fast, just go here to the playground example and set dir="rtl" for the first carousel.
https://playground.amp.dev/?url=https%3A%2F%2Fpreview.amp.dev%2Fdocumentation%2Fexamples%2Fcomponents%2Famp-carousel&format=websites&_gl=1*ip4ars*_ga*YW1wLVAwTm9TZ0taOGlkeWZRcEhhcVlhc3c.

It will only load the first image, the rest are ignored for some reason. 

## What browsers are affected?
This behavior occurs on Chrome.
Although it seems to be working fine on Firefox.

## Which AMP version is affected?
The current version where this occurs is 2005150002001, it probably existed with previous versions as well.

**Comment by caroqliu** [comment]
Hi @omar-dulaimi, thanks for filing this issue! Unfortunately this is a known issue with amp-carousel version 0.1 that we will not have the bandwidth to fix for this version.

Fortunately, amp-carousel version 0.2 does support `dir=rtl`! I'd encourage you to please try using 0.2 instead.

In the mean time I'm going to close the issue seeing as the feature will remain unsupported for 0.1 and is functioning and available for 0.2. Feel free to open a new issue if you encounter any troubles migrating to the new version that we can help you debug.

**Comment by omar-dulaimi** [comment]
Hello @caroqliu, Thanks for explaining. I updated the amp carousel version to 0.2 and it does fix the problem. But I noticed that when I scroll and reach the end, it bounces back a little!
Is that documented? Or is it a result of just updating amp carousel and not everything else?
Should I upgrade anything else? Like amp-script, amp-analytics, ampproject v0 ??

I tried to locate those files online, to read more about the topic, but it's definitely not easy. I'm not able to find them yet.

                <script
                    async
                    custom-element="amp-carousel"
                    src="https://cdn.ampproject.org/v0/amp-carousel-0.1.js"
                ></script>


                <script
                    async
                    custom-element="amp-script"
                    src="https://cdn.ampproject.org/v0/amp-script-0.1.js"
                ></script>


                <script
                    async
                    custom-element="amp-analytics"
                    src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js"
                ></script>


                <script async src="https://cdn.ampproject.org/v0.js"></script>


                <script
                    async
                    custom-element="amp-lightbox-gallery"
                    src="https://cdn.ampproject.org/v0/amp-lightbox-gallery-0.1.js"
                ></script>


                <script
                    async
                    custom-element="amp-social-share"
                    src="https://cdn.ampproject.org/v0/amp-social-share-0.1.js"
                ></script>

**Comment by caroqliu** [comment]
Could you send a video of the carousel bouncing back a little, or a repro case for this? I'm not sure I fully understand. You should not have to update any other scripts except `amp-carousel`:
`<script async custom-element="amp-carousel" src="https://cdn.ampproject.org/v0/amp-carousel-0.2.js"></script>`

**Comment by omar-dulaimi** [comment]
![captured](https://user-images.githubusercontent.com/11743389/83916737-5ca11580-a77e-11ea-9cf3-5965528c6a51.gif)

It happens with and without setting dir. If you look closely, I scroll the list to the left, then to the right. Once I finish scrolling to the right, it bounces back multiple times just a little distance.

**Comment by omar-dulaimi** [comment]
Hello @kristoferbaxter, Should I submit a new ticket since @caroqliu  is not responding?

**Comment by alanorozco** [comment]
@omar-dulaimi 

Thanks for filing this. Taking it over.

I'm trying to reproduce the bouncing issue (using 0.2) but I'm having trouble. I'm using [this code sample.](https://playground.amp.dev/?format=websites&_gl=1*ip4ars*_ga*YW1wLVAwTm9TZ0taOGlkeWZRcEhhcVlhc3c#share=PCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoT4KPGhlYWQ+CiAgPG1ldGEgY2hhcnNldD0idXRmLTgiPgogIDx0aXRsZT5hbXAtY2Fyb3VzZWw8L3RpdGxlPgogIDxzY3JpcHQgYXN5bmMgc3JjPSJodHRwczovL2Nkbi5hbXBwcm9qZWN0Lm9yZy92MC5qcyI+PC9zY3JpcHQ+CiAgPHNjcmlwdCBhc3luYyBjdXN0b20tZWxlbWVudD0iYW1wLWNhcm91c2VsIiBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwL2FtcC1jYXJvdXNlbC0wLjIuanMiPjwvc2NyaXB0PgogIDxsaW5rIHJlbD0iY2Fub25pY2FsIiBocmVmPSJodHRwczovL2FtcC5kZXYvZG9jdW1lbnRhdGlvbi9leGFtcGxlcy9jb21wb25lbnRzL2FtcC1jYXJvdXNlbC9pbmRleC5odG1sIj4KICA8bWV0YSBuYW1lPSJ2aWV3cG9ydCIgY29udGVudD0id2lkdGg9ZGV2aWNlLXdpZHRoLG1pbmltdW0tc2NhbGU9MSxpbml0aWFsLXNjYWxlPTEiPgogIDxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoOy1tb3otYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7LW1zLWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoO2FuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RofUAtd2Via2l0LWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW1vei1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1tcy1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1vLWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1Aa2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fTwvc3R5bGU+PG5vc2NyaXB0PjxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjpub25lOy1tb3otYW5pbWF0aW9uOm5vbmU7LW1zLWFuaW1hdGlvbjpub25lO2FuaW1hdGlvbjpub25lfTwvc3R5bGU+PC9ub3NjcmlwdD4KCjwvaGVhZD4KPGJvZHk+CiAgPGFtcC1jYXJvdXNlbCBoZWlnaHQ9IjMwMCIgbGF5b3V0PSJmaXhlZC1oZWlnaHQiIHR5cGU9ImNhcm91c2VsIj4KICAgIDxhbXAtaW1nIHNyYz0iL3N0YXRpYy9zYW1wbGVzL2ltZy9pbWFnZTEuanBnIiB3aWR0aD0iMjAwIiBoZWlnaHQ9IjE1MCIgYWx0PSJhIHNhbXBsZSBpbWFnZSI+PC9hbXAtaW1nPgogICAgPGFtcC1pbWcgc3JjPSIvc3RhdGljL3NhbXBsZXMvaW1nL2ltYWdlMi5qcGciIHdpZHRoPSIyMDAiIGhlaWdodD0iMTUwIiBhbHQ9ImFub3RoZXIgc2FtcGxlIGltYWdlIj48L2FtcC1pbWc+CiAgICA8YW1wLWltZyBzcmM9Ii9zdGF0aWMvc2FtcGxlcy9pbWcvaW1hZ2UzLmpwZyIgd2lkdGg9IjIwMCIgaGVpZ2h0PSIxNTAiIGFsdD0iYW5kIGFub3RoZXIgc2FtcGxlIGltYWdlIj48L2FtcC1pbWc+CiAgICA8YW1wLWltZyBzcmM9Ii9zdGF0aWMvc2FtcGxlcy9pbWcvaW1hZ2UxLmpwZyIgd2lkdGg9IjIwMCIgaGVpZ2h0PSIxNTAiIGFsdD0iYSBzYW1wbGUgaW1hZ2UiPjwvYW1wLWltZz4KICAgIDxhbXAtaW1nIHNyYz0iL3N0YXRpYy9zYW1wbGVzL2ltZy9pbWFnZTIuanBnIiB3aWR0aD0iMjAwIiBoZWlnaHQ9IjE1MCIgYWx0PSJhbm90aGVyIHNhbXBsZSBpbWFnZSI+PC9hbXAtaW1nPgogICAgPGFtcC1pbWcgc3JjPSIvc3RhdGljL3NhbXBsZXMvaW1nL2ltYWdlMy5qcGciIHdpZHRoPSIyMDAiIGhlaWdodD0iMTUwIiBhbHQ9ImFuZCBhbm90aGVyIHNhbXBsZSBpbWFnZSI+PC9hbXAtaW1nPgogIDwvYW1wLWNhcm91c2VsPgo8L2JvZHk+CjwvaHRtbD4=)

It would help us out if you provide a code sample/usable URL and the platform(s) where this reproduces (iOS, Android, Chrome Desktop, Safari Desktop, etc.)

**Comment by omar-dulaimi** [comment]
Hello @alanorozco 
I couldn't submit a reproduction back in it's time. 
I was just trying to fix the fix the rtl issue on all of our carousels of version 1. The fix worked, but arrows are behaving in weird ways.
anyways, updating to v2 produced the same bouncing effect.
so it appears that you guys haven't come across/fixed this problem yet since June; which is also weird.
